prime-dotnet-webapi/Services/EnrolleeService.cs

```diff
@@ -166,10 +166,36 @@ public async Task<IEnumerable<EnrolleeListViewModel>> GetEnrolleesAsync(Enrollee
                     .Where(e => e.GPID.StartsWith(Enrollee.PaperGpidPrefix)
                         && !_context.EnrolleeLinkedEnrolments.Any(link => link.PaperEnrolleeId == e.Id))
                 )
+                .If(!string.IsNullOrWhiteSpace(searchOptions.AssignedTo), q => q
+                    .Where(e => e.Adjudicator.IDIR == searchOptions.AssignedTo)
+                )
+                .If(searchOptions.RenewalDateRangeStart.HasValue && searchOptions.RenewalDateRangeEnd.HasValue, q => q
+                    .Where(e => e.ExpiryDate != null
+                        && e.ExpiryDate >= searchOptions.RenewalDateRangeStart
+                        && e.ExpiryDate <= searchOptions.RenewalDateRangeEnd
+                    )
+                )
+                .If(searchOptions.AppliedDateRangeStart.HasValue && searchOptions.AppliedDateRangeEnd.HasValue, q => q
+                    .Where(e => e.AppliedDate != null
+                        && e.AppliedDate >= searchOptions.AppliedDateRangeStart
+                        && e.AppliedDate <= searchOptions.AppliedDateRangeEnd
+                    )
+                )
+                .If(!string.IsNullOrWhiteSpace(searchOptions.SortOrder), q =>
+                        searchOptions.SortOrder switch
+                        {
+                            "renewalDate_asc" => q.OrderByDescending(s => s.ExpiryDate.HasValue).ThenBy(e => e.ExpiryDate).ThenBy(e => e.Id),
+                            "renewalDate_desc" => q.OrderBy(s => s.ExpiryDate.HasValue).ThenByDescending(e => e.ExpiryDate).ThenBy(e => e.Id),
+                            "appliedDate_asc" => q.OrderByDescending(s => s.AppliedDate.HasValue).ThenBy(e => e.AppliedDate).ThenBy(e => e.Id),
+                            "appliedDate_desc" => q.OrderBy(s => s.AppliedDate.HasValue).ThenByDescending(e => e.AppliedDate).ThenBy(e => e.Id),
```

Comment:
Can you add a comment re: why there is code like `q.OrderByDescending(s => s.ExpiryDate.HasValue)`?